connectors/spark/src/main/scala/io/unitycatalog/spark/CredPropsUtil.java

```diff
@@ -0,0 +1,252 @@
+      String uri,
+      String token,
+      String tableId,
+      TableOperation tableOp,
+      TemporaryCredentials tempCreds) {
+    AwsCredentials awsCred = tempCreds.getAwsTempCredentials();
+    S3PropsBuilder builder = new S3PropsBuilder()
+        .set(UCHadoopConf.S3A_CREDENTIALS_PROVIDER, AwsVendedTokenProvider.class.getName())
+        .uri(uri)
+        .token(token)
+        .uid(UUID.randomUUID().toString())
+        .credentialType(UCHadoopConf.UC_CREDENTIALS_TYPE_TABLE_VALUE)
+        .tableId(tableId)
+        .tableOperation(tableOp)
+        .set(UCHadoopConf.S3A_INIT_ACCESS_KEY, awsCred.getAccessKeyId())
+        .set(UCHadoopConf.S3A_INIT_SECRET_KEY, awsCred.getSecretAccessKey())
+        .set(UCHadoopConf.S3A_INIT_SESSION_TOKEN, awsCred.getSessionToken());
+
+    // For the static credential case, nullable expiration time is possible.
+    if (tempCreds.getExpirationTime() != null) {
+      builder.set(UCHadoopConf.S3A_INIT_CRED_EXPIRED_TIME,
+          String.valueOf(tempCreds.getExpirationTime()));
+    }
+
+    return builder.build();
+  }
+
+  public static Map<String, String> s3PathTempCredProps(
+      String uri,
+      String token,
+      String path,
+      PathOperation pathOp,
+      TemporaryCredentials tempCreds
+  ) {
+    AwsCredentials awsCred = tempCreds.getAwsTempCredentials();
+    S3PropsBuilder builder = new S3PropsBuilder()
+        .set(UCHadoopConf.S3A_CREDENTIALS_PROVIDER, AwsVendedTokenProvider.class.getName())
+        .uri(uri)
+        .token(token)
+        .uid(UUID.randomUUID().toString())
+        .credentialType(UCHadoopConf.UC_CREDENTIALS_TYPE_PATH_VALUE)
+        .path(path)
+        .pathOperation(pathOp)
+        .set(UCHadoopConf.S3A_INIT_ACCESS_KEY, awsCred.getAccessKeyId())
+        .set(UCHadoopConf.S3A_INIT_SECRET_KEY, awsCred.getSecretAccessKey())
+        .set(UCHadoopConf.S3A_INIT_SESSION_TOKEN, awsCred.getSessionToken());
+
+    // For the static credential case, null expiration time is possible.
+    if (tempCreds.getExpirationTime() != null) {
+      builder.set(UCHadoopConf.S3A_INIT_CRED_EXPIRED_TIME,
+          String.valueOf(tempCreds.getExpirationTime()));
+    }
+
+    return builder.build();
+  }
+
+  public static Map<String, String> gsProps(TemporaryCredentials tempCreds) {
+    GcpOauthToken gcpOauthToken = tempCreds.getGcpOauthToken();
+    return new GcsPropsBuilder()
+        .set(GcsVendedTokenProvider.ACCESS_TOKEN_KEY, gcpOauthToken.getOauthToken())
+        .set(GcsVendedTokenProvider.ACCESS_TOKEN_EXPIRATION_KEY,
+            String.valueOf(tempCreds.getExpirationTime()))
+        .set("fs.gs.create.items.conflict.check.enable", "false")
+        .set("fs.gs.auth.type", "ACCESS_TOKEN_PROVIDER")
+        .set("fs.gs.auth.access.token.provider", GcsVendedTokenProvider.class.getName())
+        .set("fs.gs.impl.disable.cache", "true")
+        .build();
+  }
+
+  public static Map<String, String> abfsProps(TemporaryCredentials tempCreds) {
```

Comment:
why is there no path stuff for gs and abfs? did we not support those before?

Reply:
Enabling the credential renewal for `gs` and `abfs`, is not the target of this PR. so for `gs` and `abfs`, let's keep it as it is.  The further e2e test PR will enable for `gs` and `abfs`.